### ISSUE TITLE
schema 2.0 return expected string or buffer

### ISSUE THREAD

**Issue by TomFaulkner** [p1c2u/openapi-spec-validator#26]
Without specifying `--schema 2.0` the file is read, but a parsing error occurs (see bottom), specifying 2.0 returns `expected string or buffer`. This should be a 2.0 valid file, in use with connexxion.

Is this due to an error in parsing the yaml, or is something not right when using this parameter?

```
openapi-spec-validator --schema 2.0 cms/cms/openapi/cms.yaml
expected string or buffer
```

```
'openapi' is a required property

Failed validating 'required' in schema:
```

**Comment by erunion** [comment]
I'm also seeing this on 3.0.0 specs that I'm trying to validate.

```
$ openapi-spec-validator --schema=3.0.0 docs/2.9/api.yaml 
expected string or buffer

```

**Comment by dtkav** [comment]
This is related to the response codes in the spec being parsed as integers instead of strings (which is valid yaml, but not json).
You can work around this by putting quotes around the response codes in your spec.
Related pull requests that fixes by serializing to json, then parsing the json: #18 
There's probably a much more efficient way to allow integer response codes.

**Comment by dtkav** [comment]
should be fixed by #29
